openstf/model/model_creator.py

```diff
@@ -6,21 +6,31 @@
 from lightgbm import LGBMRegressor
 
 from openstf.enums import MLModelType
+from openstf.model.xgb_quantile import XgbQuantile
 
 
 class ModelCreator:
+
     """Factory object for creating model trainer objects"""
 
     # Set object mapping
     MODEL_TRAINER_CONSTRUCTORS = {
         MLModelType.XGB: XGBRegressor,
         MLModelType.LGB: LGBMRegressor,
+        MLModelType.XGB_QUANTILE: XgbQuantile,
     }
 
     @staticmethod
-    def create_model(model_type: MLModelType) -> RegressorMixin:
+    def create_model(pj: dict) -> RegressorMixin:
```

Comment:
I'm not totally sure I like this interface better. The old interface was a bit more clean and the `prediction_job` is still a very loosely defined data structure (perhaps we can/should improve this). I guess an alternative solution would be:

```python
def create_model(model_type: MLModelType, *args, **kwargs):
  return ModelCreator.MODEL_TRAINER_CONSTRUCTORS[MLModelType(model_type)](**args, **kwargs)
```
We could in this case be more specific with the type hinting, I found this stackoverflow answer which could be helpful: https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs

Reply:
I disagree here, with the above construction you would need to pass multiple entries of the prediciton jobs as arguments (model type and quantiles). I think it is better to work on a better defenition of the prediction jobs to make the interface more clear.

Reply:
That would solve a lot of long lasting issues :) and sound like a good solution as well. Probably a better solution in general. But right now we went from a good defined interface to a vaguely defined interface.

Reply:
It does look like the current commit is not valid: 
![image](https://user-images.githubusercontent.com/6715707/122082532-5e290300-ce00-11eb-9741-e5c2e75a6ca5.png)
Not all constructors should take the quantiles argument right. And I still think the caller of the XGBQuantileRegressor should not be responsible to do this tuple conversion. We should either always use a tuple (but there are many more cases where we could use a tuple instead of a list) or just trust the developer by using plain lists or let the constructor handle this if we really think its required.